specs/language/conversions.tex

```diff
@@ -138,3 +141,39 @@
 A prvalue of type "\textit{cv1} \texttt{T}" can be converted to a prvalue of type
 "\textit{cv2} \texttt{T}" if type "\textit{cv2} \texttt{T}" is more cv-qualified
 than "\textit{cv1} \texttt{T}".
+
+\Sec{Conversion Rank}{Conv.rank}
+
+\p Every integer and floating point type have defined conversion ranks.
+A \textit{promotion} is any conversion of a value from a lower conversion rank
+to a higher conversion rank.
+
+\Sub{Integer Conversion Rank}{Conv.rank.int}
+
+\begin{itemize}
+  \item  No two signed integer types shall have the same conversion rank even if
+  they have the same representation.
+  \item The rank of a signed integer type shall be greater than the rank of any
+  signed integer type with a smaller size.
```

Comment:
curious:  What size does bool come out to be?

Reply:
Ooo... This is another area I haven't yet specified, but need to. I'll create another PR to cover that, I filed #195 to track it.

The short version of the answer is that HLSL has 16, 32, and 64-bit integer types and `bool`. `bool` is 32-bit... although we allow packing a `vector<bool,N>` into a vector of `i1` values because `bool` is defined to only be `true` or `false`, so any value other than 0 or 1 is undefined.

I'll write up that spec 😁